### PR TITLE
loader : Log upsert and remove route errors

### LINKED ISSUE #15282: Log errors on upsert/delete of endpoint routes
When configuring per-endpoint routes, we should log errors:
https://github.com/cilium/cilium/blob/fde0bdac8aa538996ce29cf8275c6f92e0663512/pkg/datapath/loader/loader.go#L348-L356

https://github.com/cilium/cilium/pull/15228 added code to remove the routes when per-endpoint routes are disabled. We should also log errors in that case, unless the error is `ENOENT` when attempting to remove a route that doesn't exist.

### PR DESCRIPTION
**Pull request by h3llix**
Fixes : #15282
Signed-off-by: Gaurav Genani <h3llix.pvt@gmail.com>

### CODE REVIEW

**Review by aanm — APPROVED** [review]
now looks great :D Thanks!

**Comment by aanm** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]

**Comment by pchaigno** [comment]
test-1.20-4.19

**Comment by pchaigno** [comment]
I opened https://github.com/cilium/cilium/issues/15572 for the new flake. Merging.

File: pkg/datapath/loader/loader.go [hunk 1]
```diff
@@ -351,18 +351,32 @@ func (l *Loader) reloadDatapath(ctx context.Context, ep datapath.Endpoint, dirs
 	}
 
 	if ip := ep.IPv4Address(); ip.IsSet() {
+		scopedLog := ep.Logger(Subsystem).WithFields(logrus.Fields{
+			logfields.Veth: ep.InterfaceName(),
+		})
 		if ep.RequireEndpointRoute() {
-			upsertEndpointRoute(ep, *ip.IPNet(32))
+			if err := upsertEndpointRoute(ep, *ip.IPNet(32)); err != nil {
+				scopedLog.WithError(err).Warn("Failed to upsert route")
```
Comment (joestringer) [inline pkg/datapath/loader/loader.go:359]: Does this log message contain the actual route which could not be added/removed? That would be super helpful details to have when these cases are hit.

**Pull request merged**